src/datamodel/builder/Payload.cpp

```diff
@@ -10,3 +10,100 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "up-cpp/datamodel/builder/Payload.h"
+namespace uprotocol::datamodel::builder {
+
+// Byte array constructor
+Payload::Payload(const std::vector<uint8_t>& value_bytes,
+                 v1::UPayloadFormat format) {
+	if (format < v1::UPayloadFormat::UPAYLOAD_FORMAT_UNSPECIFIED ||
+	    format > v1::UPayloadFormat::UPAYLOAD_FORMAT_SHM) {
+		throw std::out_of_range("Invalid byte vector payload format");
```

Comment:
Is there a way to use constructor delegation to get all this enum range validation into a single place. Barring that, shouldn't there be a private static Payload method to do this range checking, and throw the exception -or- having this type of thing in a free function in a protobuf utils namespace to centralize this stuff?

Reply:
I have moved the enum range validation in a common procedure which is then called from constructor. please let me know if that is ok or if you have any better solution.

Reply:
I think you saw and reacted to the chat about enum validators. There are '*_IsValid(int)' methods for each of the enums.

Reply:
See example here: https://github.com/eclipse-uprotocol/up-cpp/pull/158#discussion_r1644945215

Reply:
Thanks for suggestion. I have now used UPayloadFormat_IsValid(format) and it is neat and clean